### PR TITLE
Buffered Intercomponent Communication

### PR DESCRIPTION
**Pull request by niknal357**
This standard outlines the principles and mechanisms for buffered communication between components. (eg computer->computer, computer->network card, computer->screen, computer->device manager)

### CODE REVIEW

File: standards/buffered-intercomponent-communication.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+- RTR = 0: The receiver must ignore data and PP; the receiver is not prepared.
+- Receiver's Responsibility: Processing should only begin when RTR is set to 1.
+- Mid-transmission Readiness: If the receiver becomes ready during a transmission, it should momentarily pulse RTR to 1 and begin processing immediately.
+
+## Buffer Management
+- Buffered Receivers: Maintain RTR at 1 while the buffer has ample space (10+ free packets). If space drops below 10 packets, pulse RTR x times per 10 ticks (x = number of free packets) to alert the sender.
+- Non-Buffered Receivers: Pulse RTR once every 10 ticks to simulate a single-packet buffer.
+
+### Unsafe Communication
+If the sender does not awknowledge the receiver's readiness (aka performs an unsafe push), the state of the communication may be damaged. In this case, the sender can send a reset signal through the RST line to reset the communication. Upon resetting, the receiver should clear its buffer and reset its state. Further actions can be taken by the receiver or higher level protocols which is up to implementation.
```
Comment (endoddo) [inline standards/buffered-intercomponent-communication.md:42]:
```suggestion
If the sender does not acknowledge the receiver's readiness (aka performs an unsafe push), the state of the communication may be damaged. In this case, the sender can send a reset signal through the RST line to reset the communication. Upon resetting, the receiver should clear its buffer and reset its state. Further actions can be taken by the receiver or higher level protocols which is up to implementation.
```

File: standards/buffered-intercomponent-communication.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+**Notes:**
+- The 10-tick interval accounts for potential transmission and processing delays.
+- The 32-bit data packet width was chosen for compatibility with a wide range of computers, devices, and usecases.
```
Comment (endoddo) [inline standards/buffered-intercomponent-communication.md:46]:
```suggestion
- The 32-bit data packet width was chosen for compatibility with a wide range of computers, devices, and use cases.
```

**Review by endoddo — CHANGES_REQUESTED** [review]
fixed typos

File: standards/buffered-intercomponent-communication.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+- RTR (Ready to Receive): Indicates the receiving port can accept a new data packet.
+- PP (Pushing Packet): Confirms the sending port is actively transmitting a data packet. Valid data is present on the output line during this phase.
+
+- RTR = 1: The sending port may transmit a data packet. This signal, along with PP, may be brief (a single tick).
+- RTR = 0: The receiver must ignore data and PP; the receiver is not prepared.
+- Receiver's Responsibility: Processing should only begin when RTR is set to 1.
+- Mid-transmission Readiness: If the receiver becomes ready during a transmission, it should momentarily pulse RTR to 1 and begin processing immediately.
+
+## Buffer Management
+- Buffered Receivers: Maintain RTR at 1 while the buffer has ample space (10+ free packets). If space drops below 10 packets, pulse RTR x times per 10 ticks (x = number of free packets) to alert the sender.
```
Comment (endoddo) [inline standards/buffered-intercomponent-communication.md:38]: I think `pulse RTR x times oer 10 ticks` is misleading here. A single pulse would mean on then off so ten pulses would take 20 ticks. Here's a suggestion:
```suggestion
- Buffered Receivers: Maintain RTR at 1 while the buffer has ample space (10+ free packets). If space drops below 10 packets, set RTR to high for x ticks and then low again starting at every tenth tick (x = number of free packets) to alert the sender.
```

**Comment by codemaker4** [comment]
@niknal357 can you look at these changes please?

**Pull request merged**